internal/metrics/otelmetrics/factory_test.go

```diff
@@ -0,0 +1,238 @@
+}
+
+func TestCounter(t *testing.T) {
+	registry := promReg.NewPedanticRegistry()
+	factory := newTestFactory(t, registry)
+	counter := factory.Counter(metrics.Options{
+		Name: "test_counter",
+		Tags: map[string]string{"tag1": "value1"},
+	})
+	require.NotNil(t, counter)
+	counter.Inc(1)
+	counter.Inc(1)
+
+	testCounter := findMetric(t, registry, "test_counter_total")
+	metrics := testCounter.GetMetric()
+	assert.Equal(t, float64(2), metrics[0].GetCounter().GetValue())
+	expectedLabels := map[string]string{
+		"tag1":               "value1",
+		"otel_scope_name":    "jaeger-v2",
+		"otel_scope_version": "",
+	}
+	assert.Equal(t, expectedLabels, promLabelsToMap(metrics[0].GetLabel()))
+}
+
+func TestGauge(t *testing.T) {
+	registry := promReg.NewPedanticRegistry()
+	factory := newTestFactory(t, registry)
+	gauge := factory.Gauge(metrics.Options{
+		Name: "test_gauge",
+		Tags: map[string]string{"tag1": "value1"},
+	})
+	require.NotNil(t, gauge)
+	gauge.Update(2)
+
+	testGauge := findMetric(t, registry, "test_gauge")
+
+	metrics := testGauge.GetMetric()
+	assert.Equal(t, float64(2), metrics[0].GetGauge().GetValue())
+	expectedLabels := map[string]string{
+		"tag1":               "value1",
+		"otel_scope_name":    "jaeger-v2",
+		"otel_scope_version": "",
+	}
+	assert.Equal(t, expectedLabels, promLabelsToMap(metrics[0].GetLabel()))
+}
+
+func TestHistogram(t *testing.T) {
+	registry := promReg.NewPedanticRegistry()
+	factory := newTestFactory(t, registry)
+	histogram := factory.Histogram(metrics.HistogramOptions{
+		Name: "test_histogram",
+		Tags: map[string]string{"tag1": "value1"},
+	})
+	require.NotNil(t, histogram)
+	histogram.Record(1.0)
+
+	testHistogram := findMetric(t, registry, "test_histogram")
+
+	metrics := testHistogram.GetMetric()
+	assert.Equal(t, float64(1), metrics[0].GetHistogram().GetSampleSum())
+	expectedLabels := map[string]string{
+		"tag1":               "value1",
+		"otel_scope_name":    "jaeger-v2",
+		"otel_scope_version": "",
+	}
+	assert.Equal(t, expectedLabels, promLabelsToMap(metrics[0].GetLabel()))
+}
+
+func TestTimer(t *testing.T) {
+	registry := promReg.NewPedanticRegistry()
+	factory := newTestFactory(t, registry)
+	timer := factory.Timer(metrics.TimerOptions{
+		Name: "test_timer",
+		Tags: map[string]string{"tag1": "value1"},
+	})
+	require.NotNil(t, timer)
+	timer.Record(100 * time.Millisecond)
+
+	testTimer := findMetric(t, registry, "test_timer_seconds")
+
+	metrics := testTimer.GetMetric()
+	assert.Equal(t, float64(0.1), metrics[0].GetHistogram().GetSampleSum())
+	expectedLabels := map[string]string{
+		"tag1":               "value1",
+		"otel_scope_name":    "jaeger-v2",
+		"otel_scope_version": "",
+	}
+	assert.Equal(t, expectedLabels, promLabelsToMap(metrics[0].GetLabel()))
+}
+
+func TestNamespace(t *testing.T) {
+	testCases := []struct {
+		nsOptions      metrics.NSOptions
+		expectedName   string
+		expectedLabels map[string]string
+	}{
+		{
```

Comment:
you are not doing table drive test correctly, as illustrated by L107 of the factory still not covered. The point of the table tests is to run through _different use cases_, e.g. "factory with a single namespace" vs. "factory with nested namespaces". All you're doing is passing different values under the same conditions, i.e. exercising just a single use case.